drivers/serial/uart_gecko.c

```diff
@@ -488,6 +492,33 @@ static int uart_gecko_init(const struct device *dev)
 	return 0;
 }
 
+#ifndef CONFIG_PM_DEVICE
+#define GECKO_USART_PM_INST(idx) NULL
+#else
+#define GECKO_USART_PM_INST(idx) PM_DEVICE_DT_INST_GET(idx)
```

Comment:
`PM_DEVICE_DT_INST_GET()` already return NULL if `!CONFIG_PM_DEVICE`. This macro is not needed.